relation/src/gadgets/cmp.rs

```diff
@@ -316,4 +419,92 @@ mod test {
         }
         Ok(())
     }
+    fn test_is_le_constant<F: PrimeField>(a: &F, b: &F) -> Result<(), CircuitError> {
+        let mut circuit = PlonkCircuit::<F>::new_turbo_plonk();
+        let expected_result = if a < b { F::one() } else { F::zero() };
+        let a = circuit.create_variable(*a)?;
+
+        let c = circuit.is_lt_constant(a, *b)?;
+        assert!(circuit.witness(c.into())?.eq(&expected_result));
+        assert!(circuit.check_circuit_satisfiability(&[]).is_ok());
+        Ok(())
+    }
+    fn test_is_leq_constant<F: PrimeField>(a: &F, b: &F) -> Result<(), CircuitError> {
```

Comment:
can we unify these to a single function instead?
`test_cmp(a: &F, b: &F, order: Ordering, also_check_equality: bool, is_b_constant: bool)`

Reply:
Agree, now I also see the advantage of having a unified API `is_cmp(...)` and `enforce_cmp()` as it can remove code repetition. Not sure if we want to change the APIs, but agree that we should do it at least in tests.